libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -45,204 +45,303 @@ class AC_PosControl
+    // Runs the NE-axis position controller, computing output acceleration from position and velocity errors.
+    // Uses P and PID controllers to generate corrections which are added to feedforward velocity/acceleration.
+    // Requires all desired targets to be pre-set using the input_* or set_* methods.
     void update_NE_controller();
 
     ///
     /// Vertical position controller
     ///
 
-    /// set_max_speed_accel_U_cmss - set the maximum vertical speed in cm/s and acceleration in cm/s/s
-    ///     speed_down_cms may be positive or negative, but it is always interpreted as a descent rate.
-    ///     This can be done at any time as changes in these parameters are handled smoothly
-    ///     by the kinematic shaping.
+    // Sets maximum climb/descent rate (cm/s) and vertical acceleration (cm/s²) for the U-axis.
+    // Descent rate may be positive or negative and is always interpreted as a descent.
+    // See set_max_speed_accel_U_m() for full details.
     void set_max_speed_accel_U_cm(float speed_down_cms, float speed_up_cms, float accel_cmss);
 
-    /// set_correction_speed_accel_U_cmss - set the position controller correction velocity and acceleration limit
-    ///     speed_down_cms may be positive or negative, but it is always interpreted as a descent rate.
-    ///     This should be done only during initialisation to avoid discontinuities
+    // Sets maximum climb/descent rate (m/s) and vertical acceleration (m/s²) for the U-axis.
+    // These values are used for jerk-limited kinematic shaping of the vertical trajectory.
+    void set_max_speed_accel_U_m(float speed_down_ms, float speed_up_ms, float accel_mss);
+
+    // Sets vertical correction velocity and acceleration limits (cm/s, cm/s²).
+    // Should only be called during initialization to avoid discontinuities.
+    // See set_correction_speed_accel_U_mss() for full details.
     void set_correction_speed_accel_U_cmss(float speed_down_cms, float speed_up_cms, float accel_cmss);
 
-    /// get_max_accel_U_cmss - get the maximum vertical acceleration in cm/s/s
-    float get_max_accel_U_cmss() const { return _accel_max_u_cmss; }
+    // Sets vertical correction velocity and acceleration limits (m/s, m/s²).
+    // These values constrain the correction output of the PID controller.
+    void set_correction_speed_accel_U_mss(float speed_down_ms, float speed_up_ms, float accel_mss);
+
+    // Returns maximum vertical acceleration in cm/s².
+    // See get_max_accel_U_mss() for full details.
+    float get_max_accel_U_cmss() const { return get_max_accel_U_mss() * 100.0; }
+
+    // Returns maximum vertical acceleration in m/s² used for shaping the climb/descent trajectory.
+    float get_max_accel_U_mss() const { return _accel_max_u_mss; }
+
+    // Returns maximum allowed positive (upward) position error in cm.
+    // See get_pos_error_up_m() for full details.
+    float get_pos_error_up_cm() { return get_pos_error_up_m() * 100.0; }
```

Comment:
again a hang-over from existing code but these four functions can be made const:

float get_pos_error_up_cm() const
float get_pos_error_up_m() const
float get_pos_error_down_cm() const
float get_pos_error_down_m() const